lib/libc/sys/select.c

```diff
@@ -0,0 +1,56 @@
+#include <sys/select.h>
+#include <sys/time.h>
+#include <sys/event.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <errno.h>
+
+int select(int nfds, fd_set *restrict readfds, fd_set *restrict writefds,
+           fd_set *restrict exceptfds, struct timeval *restrict timeout) {
+    int kq;
+    int ret;
+    struct timespec timeout_ts;
+    struct kevent *events;
+    int nevents = 0;
+
+    if (timeout != NULL)
```

Comment:
Check timeout too.

Reply:
Could you elaborate on that? What should I check? The specification of `select` syscall allows all values of `timeout`.

Reply:
I was thinking about returning `EINVAL` for negative and large values of timeout.

Reply:
I see. The specification however, doesn't say anything about big nor negative values.

Reply:
NetBSD's [select](https://man.netbsd.org/select.2) manual states:

> [EINVAL] The specified time limit is invalid.  One of its components is negative or too large.

Reply:
This is interesting. I have no idea where I found that information a few months ago.